### PR TITLE
Add markdown preview

### PR DESCRIPTION
**Pull request by alex-ju**
This PR:
 - Makes the markdown editor a component
 - Allows markdown preview
 - Sets `$govuk-global-styles: true;` to style anchors, paragraphs and headings in markdown preview

### CODE REVIEW

File: app/assets/javascripts/components/markdown-editor.js [hunk 1]
```diff
@@ -1,2 +1,70 @@
...
+
+  var $preview = this.$preview
+  var text = this.$input.value
+
+  // Mirror textarea's height
+  $preview.style.height = this.$input.offsetHeight + 'px'
+
+  // Render markdown
+  window.marked(text, function (err, content) {
+    if (err) throw err
```
Comment (benthorner) [inline app/assets/javascripts/components/markdown-editor.js]: What kind of errors are we anticipating here - should we be handling them?

File: app/assets/javascripts/components/markdown-editor.js [hunk 2]
```diff
@@ -1,2 +1,70 @@
...
+
+MarkdownEditor.prototype.toggleElements = function () {
+  this.toggle(this.$editButton)
+  this.toggle(this.$previewButton)
+  this.toggle(this.$editorInput)
+  this.toggle(this.$previewBody)
+}
+
+MarkdownEditor.prototype.toggle = function (element) {
+  if (element.ownerDocument.defaultView.getComputedStyle(element, null).display === 'none') {
```
Comment (benthorner) [inline app/assets/javascripts/components/markdown-editor.js]: Would element.style.display not work? Might be good to document why we need the computed version.

File: app/assets/javascripts/components/markdown-editor.js (hunk 1 above)
Comment (alex-ju) [reply app/assets/javascripts/components/markdown-editor.js]: marked can return parsing errors and if they happen I'm thinking we don't want them to fail silently.

Comment (benthorner) [reply app/assets/javascripts/components/markdown-editor.js]: Totally agree we shouldn't forget about the error, but how will the user know they've done something wrong?

File: app/assets/javascripts/components/markdown-editor.js (hunk 2 above)
Comment (alex-ju) [reply app/assets/javascripts/components/markdown-editor.js]: we need `getComputedStyle` to evaluate the styles coming from CSS. `element.style.display` only reads styles applied with JavaScript.

File: app/assets/javascripts/components/markdown-editor.js (hunk 1 above)
Comment (alex-ju) [reply app/assets/javascripts/components/markdown-editor.js]: They're evaluating the preview coming from content, but they may not realise that indeed. I added 'Error previewing content' as a placeholder if marked throws an error.

**Pull request merged**